src/plugins/wizard/server/saved_objects/wizard_app.ts

```diff
@@ -47,6 +47,9 @@ export const wizardSavedObjectType: SavedObjectsType = {
         index: false,
       },
       version: { type: 'integer' },
+      kibanaSavedObjectMeta: {
```

Comment:
Agreed - and if we need to add it here now, it's crying out for an explanatory comment about why it's still needed and which other plugins/services rely on it.

Reply:
When we save a saved object, the save object class(in saved objects plugin) will call build saved object function as well as serialize saved object function. Within the serialized function, the extractReference() will extract search source into two parts. The first part will include the index ref name and the function will create a kibanaSavedObjectMeta attribute to store it; the second part will be a reference object that includes the index ref name, index pattern type and index pattern id, and the function will create a separate reference attribute and push this reference object into the reference array. 

When loading the saved object, the saved object class will then get the index information from the reference object, and inject the information back into the kibanaSavedObjectMeta, in the injectReference() function.

It is still needed because the other plugins (such as visualization plugin, dashboard plugin) that depends on saved object plugin still rely on these services and this type of saved object structure. Maybe in the future we can rename this "kibanaSavedObjectMeta"?

Here is the logic that locates in serialize_saved_object.ts:
```
if (savedObject.searchSourceFields) {
    const [searchSourceFields, searchSourceReferences] = extractSearchSourceReferences(
      savedObject.searchSourceFields
    );
    const searchSourceJSON = JSON.stringify(searchSourceFields);
    attributes.kibanaSavedObjectMeta = { searchSourceJSON };
    references.push(...searchSourceReferences);
  }
```

For example, a visualization saved object: 
```
"visualization" : {
            "title" : "[Logs] Visitors by OS",
            "visState" : "{"title":"[Logs] Visitors by ... ... ",
            ... ...
            "kibanaSavedObjectMeta" : {
              "searchSourceJSON" : """{"filter":[],"query":{"query":"","language":"kuery"},"indexRefName":"kibanaSavedObjectMeta.searchSourceJSON.index"}"""
            }
          },
          "type" : "visualization",
          "references" : [
            {
              "name" : "kibanaSavedObjectMeta.searchSourceJSON.index",
              "type" : "index-pattern",
              "id" : "90943e30-9a47-11e8-b64d-95841ca0b247"
            }
          ],
        }

```

Reply:
I can add a brief explanatory comment, and i will document more here https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2205 ?